### PR TITLE
LRAC-6760 Prevent tracking if is in edit mode

### PR DESCRIPTION
**Pull request by fcodias**
Jira issue: https://issues.liferay.com/browse/LRAC-6760

Hey guys this PR is to avoid firing events when user is editing a page or in a configuration portlet.

I could figure out in DXP's code base [this function in FragmentRendererUtil](https://github.com/liferay/liferay-portal/blob/8d67629d476771a6597de04a3bd1b106b228927f/modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentRendererUtil.java#L70) but as it is intern module so I made a copy/paste from there and updated to fire only if it's in `view` mode, this adjustment was needed because there is an iframe in `preview` mode when the user is editing the page.

I couldn't get the `portletId` using the `portletDisplay` from the `themeDisplay`, so I made it in the same pattern as before. I could see that there is a Util related to the [portletConfiguration name ](https://github.com/liferay/liferay-portal/blob/8d67629d476771a6597de04a3bd1b106b228927f/modules/apps/portlet-configuration/portlet-configuration-web/src/main/java/com/liferay/portlet/configuration/web/internal/constants/PortletConfigurationPortletKeys.java#L22) but it is intern too, so I copied in the same pattern from there.
I could [see some usages in the code base](https://github.com/liferay/liferay-portal/search?q=%22com_liferay_portlet_configuration_web_portlet_%22&type=code) in the same way so I think it's ok.

You guys feel free to change or request anything as I'm not sure if it was the right decision, or even if there is an anti-pattern.

Thanks!

cc @marcellustavares @marcosapmf

### CODE REVIEW

File: modules/apps/analytics/analytics-web/src/main/java/com/liferay/analytics/web/internal/servlet/taglib/AnalyticsTopHeadJSPDynamicInclude.java [hunk 1]
```diff
@@ -217,6 +222,21 @@ private boolean _isAnalyticsTrackingEnabled(
 		return false;
 	}
 
+	private boolean _isConfigurationPortlet(
```
Comment (andreldm) [inline modules/apps/analytics/analytics-web/src/main/java/com/liferay/analytics/web/internal/servlet/taglib/AnalyticsTopHeadJSPDynamicInclude.java]: Note to myself: check if `Portal#isControlPanelPortlet` works.

File: modules/apps/analytics/analytics-web/src/main/java/com/liferay/analytics/web/internal/servlet/taglib/AnalyticsTopHeadJSPDynamicInclude.java [hunk 2]
```diff
@@ -237,6 +242,17 @@ private boolean _isConfigurationPortlet(
 		return false;
 	}
 
+	private boolean _isEditMode(HttpServletRequest httpServletRequest) {
+		String layoutMode = ParamUtil.getString(
+			httpServletRequest, "p_l_mode", Constants.VIEW);
+
+		if (!layoutMode.equals(Constants.VIEW)) {
```
Comment (marcellustavares) [inline modules/apps/analytics/analytics-web/src/main/java/com/liferay/analytics/web/internal/servlet/taglib/AnalyticsTopHeadJSPDynamicInclude.java]: I think you can't assume that it's in EDIT mode when it's not VIEW. I'm pretty sure there are more modes, see https://liferayiseasy.blogspot.com/2015/09/portlet-modes-in-liferay.html

**Comment by andreldm** [comment]
@fcodias as we have discussed, this fix doesn't solve the problem when SPA is enabled.

**Comment by andreldm** [comment]
https://github.com/fcodias/liferay-portal/pull/28

**Pull request closed without merging**